src/main/java/taxi/dao/jdbcimpl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,210 @@
+        String newLinkQuery = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateStatement = connection.prepareStatement(updateQuery);
+                 PreparedStatement unlinkStatement = connection.prepareStatement(unlinkQuery);
+                 PreparedStatement newLinkStatements = connection.prepareStatement(newLinkQuery)) {
+
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+
+            unlinkStatement.setLong(1, car.getId());
+            unlinkStatement.executeUpdate();
+
+            newLinkStatements.setLong(2, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                newLinkStatements.setLong(1, driver.getId());
+                newLinkStatements.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not update car: "
+                    + car.toString(), e);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars c"
+                + " SET deleted = TRUE"
+                + " WHERE c.id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteStatement = connection.prepareStatement(deleteQuery)) {
+            deleteStatement.setLong(1, id);
+            int affectedRows = deleteStatement.executeUpdate();
+            return affectedRows != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not delete car with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT c.id AS car_id, model, "
+                + "m.manufacturer_name AS manufacturer_name, "
+                + "m.manufacturer_country AS manufacturer_country, "
+                + "m.id AS manufacturer_id "
+                + "FROM cars_drivers "
+                + "JOIN cars c ON c.id = cars_drivers.car_id "
+                + "JOIN drivers d ON d.id = cars_drivers.driver_id "
+                + "JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE c.deleted = FALSE AND d.deleted = FALSE "
+                + "AND d.id = ?;";
+        List<Car> carList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(createCar(resultSet, connection));
+            }
```

Comment:
same comments as for `getAll` and `get` methods

Reply:
Done as well